cfme/cloud/keypairs.py

```diff
@@ -119,19 +119,54 @@ def is_displayed(self):
     form = View.nested(KeyPairAddForm)
 
 
-class KeyPair(Taggable, SummaryMixin, Navigatable):
+class KeyPairCollection(Navigatable):
+    """ Collection object for the :py:class: `cfme.cloud.KeyPair`. """
+
+    def create(self, name, provider, public_key=None, cancel=False):
+        """Create new keyPair.
+
+        Args:
+            name (str): name of the KeyPair
+            public_key (str): RSA Key if present
+            provider (str): Cloud Provider
+        """
+
+        view = navigate_to(self, 'Add')
+        view.form.fill({'name': name, 'public_key': public_key, 'provider': provider.name})
+        if cancel:
+            view.form.cancel.click()
+            flash_message = 'Add of new Key Pair was cancelled by the user'
+        else:
+            view.form.add.click()
+            flash_message = VersionPick({
+                Version.lowest(): 'Creating Key Pair {}'.format(name),
+                '5.8': 'Key Pair "{}" created'.format(name)}).pick(self.appliance.version)
+
+        # add/cancel should redirect, new view
+        view = self.create_view(KeyPairAllView)
+        # TODO BZ 1444520 causing ridiculous redirection times after submitting the form
+        wait_for(lambda: view.is_displayed, num_sec=120, delay=3,
+                 fail_func=lambda: view.flush_widget_cache(), handle_exception=True)
```

Comment:
Don't need a lambda for `fail_func` since you're passing a bound method, you can pass `view.flush_widget_cache`